### PR TITLE
fix: add new notification settings frontend

### PR DESCRIPTION
**Pull request by rebelchris**
### Describe what this PR does
- Related backend: https://github.com/dailydotdev/daily-api/pull/1932
- Figma: https://www.figma.com/design/nmfWPS7x3kzLUvYMkBx2kW/daily.dev---Dev-Mode?node-id=9439-24492&m=dev
- Streaks is out of scope
- TODO: Analytics seem a bit off, need to sync with Ido I think. 

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

AS-339 AS-340 #done

### Preview domain
https://as-340-notification-settings.preview.app.daily.dev

### CODE REVIEW

File: packages/webapp/pages/account/notifications.tsx [hunk 1]
```diff
@@ -40,10 +49,21 @@ const AccountNotificationsPage = (): ReactElement => {
...
-    isLoading: isPersonalizedDigestLoading,
+    readingReminder,
     subscribePersonalizedDigest,
     unsubscribePersonalizedDigest,
   } = usePersonalizedDigest();
+  const [digestTimeIndex, setDigestTimeIndex] = useState(8);
+  if (personalizedDigest?.preferredHour !== digestTimeIndex) {
+    setDigestTimeIndex(personalizedDigest?.preferredHour);
+  }
+  const [readingTimeIndex, setReadingTimeIndex] = useState(
```
Comment (nensidosari) [inline packages/webapp/pages/account/notifications.tsx:60]: I do think declaring all state on top then doing manipulations to it is better when u read the code.

**Review by nensidosari — APPROVED** [review]
Looks good, let me know if u'll do any changes to the events

File: packages/shared/src/hooks/usePersonalizedDigest.ts [hunk 2]
```diff
@@ -109,6 +156,7 @@ export const usePersonalizedDigest = (): UsePersonalizedDigest => {
 
   return {
     personalizedDigest,
+    readingReminder,
```
Comment (capJavert) [inline packages/shared/src/hooks/usePersonalizedDigest.ts:159]: I am not sure it is a good idea to split the properties in return. Basically if we ever add additional type we have to then again add property and filter for that type. I would rather the consumer of the hook map and take the digest type they need. We can even expose a function for that from the hook.

Comment (rebelchris) [reply packages/shared/src/hooks/usePersonalizedDigest.ts:159]: Happy to make it a func you can call by passing type?
This felt easier from consumer point, but get your remarks

**Comment by rebelchris** [comment]
@capJavert Modified with your feedback.
Also modified the mutations as they could be optimized a bit.

**Comment by rebelchris** [comment]
@nensidosari // @capJavert Sorry requesting again.
Finalised the tracking events now.

Basically we added tracking for both enable and disable (*previously only disable)

File: packages/webapp/pages/account/notifications.tsx [hunk 3]
```diff
@@ -52,13 +84,26 @@ const AccountNotificationsPage = (): ReactElement => {
...
 
-    if (!value) {
+    const defaultTrackingProps = {
+      extra: JSON.stringify({
+        channel: NotificationChannel.Email,
+        category: [
+          NotificationCategory.Product,
+          NotificationCategory.Marketing,
+          NotificationCategory.Digest,
+        ],
```
[the remark this answers is not in the document]
Comment (rebelchris) [reply packages/webapp/pages/account/notifications.tsx:94]: Yes exactly :)

[the remark this answers is not in the document]
Comment (rebelchris) [reply packages/webapp/pages/account/notifications.tsx:94]: and just more explicit anyway lol

[the remark this answers is not in the document]
Comment (capJavert) [reply packages/webapp/pages/account/notifications.tsx:94]: yep!

**Pull request merged**